runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/windmill/work/refresh/ActiveWorkRefresher.java

```diff
@@ -99,7 +103,7 @@ public void stop() {
   private void invalidateStuckCommits() {
```

Comment:
this is now pretty abstract as both the computation supplier and the invalidation function are injected.  Could it be a single function invalidateStuckCommits(deadline)?

Reply:
Or alternativley since this is just to get a hook for testing, can you modify the test instead to use the executor or computation supplier instead to enforce it runs?  It seems like you then call stop() which would block if the invalidation was running, either could be sufficient since it indicates it has started.

Reply:
removed and spyed in test